src/components/audio-player-controls/audio-player-controls.tsx

```diff
@@ -21,17 +21,56 @@ interface AudioPlayerControlsProps {
 const AudioPlayerControls = ({pluginConfig, player, onPluginsControlClick, showMorePluginsIcon}: AudioPlayerControlsProps) => {
   const playlist = useSelector((state: any) => state.engine.playlist);
   const ref = useRef<HTMLDivElement>();
+  const [playbackRateState, setPlaybackRateState] = useState(player.playbackRate);
+  const currentPlayerRef = useRef<HTMLDivElement>(null);
 
   useEffect(() => {
     ref.current?.setAttribute('tabindex', '0');
   }, []);
+  
+  useEffect(() => {
+    const handleKeydown = (event: KeyboardEvent) => {
+      if (!currentPlayerRef.current) return;
+  
+      const isInsidePlayer = event.target instanceof Node && currentPlayerRef.current.contains(event.target);
+      if (!isInsidePlayer) return;
+  
+      const rates = player.playbackRates;
+      const index = rates.indexOf(player.playbackRate); 
+  
+      if (event.shiftKey) {
+        let newRate = player.playbackRate;
+        switch (event.code) {
+          case 'Period':
+            if (index < rates.length - 1) newRate = rates[index + 1];
+            break;
+          case 'Semicolon':
+            newRate = player.defaultPlaybackRate;
+            break;
+          case 'Comma':
+            if (index > 0) newRate = rates[index - 1];
+            break;
+          default:
+            return;
+        }
+  
+        player.playbackRate = newRate;
+        setPlaybackRateState(newRate);
+      }
+    };
+  
+    document.addEventListener('keydown', handleKeydown);
+    return () => {
+      document.removeEventListener('keydown', handleKeydown);
```

Comment:
if you use withEventManager, the eventManager already has code to clean up the listener on unmount